data-processing/common/parse_equation.py

```diff
@@ -64,23 +66,54 @@ def get_symbols(mathml: str) -> List[Symbol]:
             if index is not None:
                 characters.append(index)
 
-        # Clean the MathML of our annotations. Do this cleaning on a clone of the node, so that the
-        # annotations are available for processing later nodes.
-        node_clone = copy.copy(symbol_node)
-        _remove_s2_annotations(node_clone)
-        for descendant in node_clone.descendants:
-            _remove_s2_annotations(descendant)
+        # Group consecutive character symbols only if they are a child of the mrow tag.
+        # The .has_attr('s2:end') is necessary as not every symbol is annotated by katex
+        if symbol_node.has_attr('s2:end') and symbol_node.parent.name in MERGABLE_PARENT_TAGS and symbol_node.name in CHARACTER_TAGS:
+            if len(consecutive_chars) > 0 and not consecutive_chars[-1]['s2:end'] == symbol_node['s2:start']:
+                node_symbols.append(_merge_consecutive_symbols(consecutive_chars, all_indices, soup.new_tag('mi')))
+                consecutive_chars = []
+                all_indices = []
+            consecutive_chars.append(symbol_node)
+            all_indices.extend(characters)
+        elif len(consecutive_chars) > 0:
+            node_symbols.append(_merge_consecutive_symbols(consecutive_chars, all_indices, soup.new_tag('mi')))
+            consecutive_chars = []
+            all_indices = []
+
+        if len(consecutive_chars) == 0:
+            # Clean the MathML of our annotations. Do this cleaning on a clone of the node, so that the
+            # annotations are available for processing later nodes.
+            node_clone = _clean_node_of_annotations(symbol_node)
+            node_symbols.append(
+                # set node to 'symbol_node', not clone, as the node must have a parent for the
+                # 'set_children' method. The clone will have had its parent removed.
+                NodeSymbol(characters=characters, mathml=str(node_clone), node=symbol_node)
+            )
 
-        node_symbols.append(
-            # set node to 'symbol_node', not clone, as the node must have a parent for the
-            # 'set_children' method. The clone will have had its parent removed.
-            NodeSymbol(characters=characters, mathml=str(node_clone), node=symbol_node)
-        )
+    # Possibility that the last symbol was part of a continuing symbol
+    if len(consecutive_chars) > 0:
```

Comment:
@zkirby isn't this redundant with Line 78?

Reply:
There is the possibility that the last symbol was also included in consecutive_chars (i.e. went through the if statement at line 71). In that case we need to clean up consecutive_chars and build anything that could be in it.